### ISSUE TITLE
Star routes seem to get priority over other routes which have trailing '/'

### ISSUE THREAD

**Issue by workmanw** [tildeio/route-recognizer#82]
Original issue in Ember: https://github.com/emberjs/ember.js/issues/13211

I have a failing test for route-recognizer, I'll open a PR in just a moment. There is an ember working demo  on the original issue.

I don't yet fully understand the mechanics of route-recognizer, but it seems that starting with 0.1.6 (and later) when you have a route that has just a trailing '/' as it's path, star routes end up prioritized over it. For example:

```
router.add([
  { path: "/foo/:dynamic", handler: handler1 }, 
  { path: "/baz/:dynamic", handler: handler2 }, 
  { path: "/", handler: handler3 }
]);
router.add([
  { path: "/foo/:dynamic", handler: handler1 }, 
  { path: "/*wildcard", handler: handlerWildcard }
]);

// Returns the `/*wildcard` route, rather than the above route.
router.recognize("/foo/r3/baz/w10"); 
```

Edit: Failing Test: #83

**Comment by jmeas** [comment]
May be related to this commit: 0499446f80c649a2df64ced5cc5458742a57c423

I'm a bit busy in life atm but I added it to my to-do list to look more into this when I've got time. Thanks for raising the issue!

**Comment by workmanw** [comment]
@jmeas Yeap, I can confirm that that is the culprit.

**Comment by jmeas** [comment]
Thanks for verifying @workmanw . Do you know when that ember version is scheduled to be released? I'd like to get it fixed in time for that!

**Comment by workmanw** [comment]
I don't. I could make a guess ... but it'd just be a guess. Maybe @rwjblue could provide a better timeline.

**Comment by jmeas** [comment]
I have a suspicion that it's related to this line:

https://github.com/tildeio/route-recognizer/blob/master/lib/route-recognizer.js#L134

which is the line I was least certain of when updating the algorithm. The tests passed here, so I figured it *might* be okay :p 

My first idea is to just delete that line. In that case, empty segments wouldn't contribute to the specificity at all. I'll need to think more about the implications of that,  though.. I'm on my phone atm so I can't do further testing.

**Comment by rwjblue** [comment]
> Do you know when that ember version is scheduled to be released? 

https://github.com/tildeio/route-recognizer/commit/0499446f80c649a2df64ced5cc5458742a57c423 (included since 0.1.6 of this lib).  Ember updated from 0.1.5 to 0.1.9 in https://github.com/emberjs/ember.js/pull/13104 which will be used in Ember 2.6.0-beta.1 (when we release it) sometime next week.

**Comment by rwjblue** [comment]
I submitted https://github.com/emberjs/ember.js/pull/13286 to downgrade route-recognizer to 0.1.5 (which is what is used in Ember 2.4/2.5) until we can get this fixed.

**Comment by workmanw** [comment]
@jmeas I tried commenting out the suspecting line in my test app, https://github.com/tildeio/route-recognizer/blob/master/lib/route-recognizer.js#L134, but it still seems busted :(. I'm guessing you thought there was more to it, but I just wanted to follow back up.

**Comment by jmeas** [comment]
np @workmanw the issue should be fixed in #84
